src/numberparse/approx.rs

```diff
@@ -437,7 +437,11 @@ impl<'de> Deserializer<'de> {
                 // FIXME
                 // can we omit this: buf.len() - byte_count >= 8
 
-                if is_made_of_eight_digits_fast(unsafe { buf.get_kinda_unchecked(byte_count..) }) {
+                if is_made_of_eight_digits_fast(unsafe {
+                    buf.get_kinda_unchecked(byte_count..)
+                        .try_into()
+                        .unwrap_unchecked()
```

Comment:
We can only `unwrap_unchecked` here if we are absolutely sure that we have 8 bytes. Are we?